core/serialisation/src/main/java/uk/gov/gchq/gaffer/serialisation/ToBytesSerialiser.java

```diff
@@ -51,13 +51,31 @@
     byte[] serialise(final T object) throws SerialisationException;
 
     /**
+     * @param allBytes The bytes to be decoded into characters
+     * @param offset   The index of the first byte to decode
+     * @param length   The number of bytes to decode
+     * @return T the deserialised object
+     * @throws SerialisationException issues during deserialisation
+     */
+    default T deserialise(final byte[] allBytes, final int offset, final int length) throws SerialisationException {
+        final byte[] selection = new byte[length];
+        try {
+            System.arraycopy(allBytes, offset, selection, 0, length);
+        } catch (NullPointerException e) {
```

Comment:
It's convention to use `final` here, i.e. `catch (final NullPointerException e)`